app/src/main/java/com/github/swent/echo/viewmodels/event/EventViewModel.kt

```diff
@@ -29,30 +43,77 @@ class EventViewModel() : ViewModel() {
         )
     private val _event = MutableStateFlow<Event>(emptyEvent)
     private val _status = MutableStateFlow<EventStatus>(EventStatus.New)
+    private val allTagsList = MutableStateFlow<List<Tag>>(listOf())
 
-    // placeholder for the list of all tags
-    private val allTagsList = listOf(Tag("1", "tag1"), Tag("2", "tag2"), Tag("3", "tag3"))
+    // initialize async values
+    init {
+        _status.value = EventStatus.Saving // avoid saving before initialization
+        viewModelScope.launch {
+            val userid = authenticationService.getCurrentUserID()
+            if (userid == null) {
+                _status.value = EventStatus.Error(R.string.event_creation_error_not_logged_in)
+                Log.e("create event", "the user is not logged in")
+            } else {
+                _event.value =
+                    _event.value.copy(
+                        creatorId = userid,
+                        organizerId = userid,
+                        organizerName = repository.getUserProfile(userid).name
+                    )
+                _status.value = EventStatus.New
+            }
+            allTagsList.value = repository.getAllTags()
+        }
+    }
 
-    // return the event
-    fun getEvent(): Event {
-        return _event.asStateFlow().value
+    // constructor for an already existing event
+    constructor(
+        repository: Repository,
+        authenticationService: AuthenticationService,
+        eventId: String
+    ) : this(repository, authenticationService) {
+        _status.value = EventStatus.Saving // avoid saving before initialization
+        viewModelScope.launch {
+            _event.value = repository.getEvent(eventId)
+            _status.value = EventStatus.Saved
+        }
     }
 
-    // return the organizer name of the event
-    fun getOrganizerName(): String {
-        // TODO: get organizer name from repository
-        return ""
+    // return the event
+    fun getEvent(): StateFlow<Event> {
```

Comment:
This could be simply a field, i.e. `val event = _event.asStateFlow()`. Then in the composable you will be able to use the value of the `StateFlow` to automatically trigger the recomposition. Otherwise, if you simply use the `Event` in you composable, it won't automatically trigger the recomposition...